src/Modules/Payments/Application/Subscriptions/RenewSubscription/SubscriptionRenewedEnqueueEmailConfirmationHandler.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CompanyName.MyMeetings.BuildingBlocks.Application.Data;
+using CompanyName.MyMeetings.Modules.Payments.Application.Configuration.Commands;
+using CompanyName.MyMeetings.Modules.Payments.Application.Payers.GetPayer;
+using CompanyName.MyMeetings.Modules.Payments.Application.Subscriptions.SendSubscriptionRenewalConfirmationEmail;
+using CompanyName.MyMeetings.Modules.Payments.Domain.Subscriptions;
+using Dapper;
+using MediatR;
+
+namespace CompanyName.MyMeetings.Modules.Payments.Application.Subscriptions.RenewSubscription
+{
+    public class SubscriptionRenewedEnqueueEmailConfirmationHandler : INotificationHandler<SubscriptionRenewedNotification>
+    {
+        private readonly ICommandsScheduler _commandsScheduler;
+        private readonly ISqlConnectionFactory _sqlConnectionFactory;
+
+        public SubscriptionRenewedEnqueueEmailConfirmationHandler(
+            ICommandsScheduler commandsScheduler,
+            ISqlConnectionFactory sqlConnectionFactory)
+        {
+            _commandsScheduler = commandsScheduler;
+            _sqlConnectionFactory = sqlConnectionFactory;
+        }
+        
+        public async Task Handle(SubscriptionRenewedNotification notification, CancellationToken cancellationToken)
+        {
+            var payer = await GetPayerDetails(notification.DomainEvent.PayerId);
+            
+            await _commandsScheduler.EnqueueAsync(new SendSubscriptionRenewalConfirmationEmailCommand(
+                Guid.NewGuid(),
+                new SubscriptionId(notification.DomainEvent.SubscriptionId),
+                payer.Email));
+        }
+        
+        private async Task<PayerDto> GetPayerDetails(Guid payerId)
+        {
+            var connection = _sqlConnectionFactory.GetOpenConnection();
+
+            const string sql = "SELECT " +
+                               "[Payer].[Id], " +
+                               "[Payer].[Login], " +
+                               "[Payer].[Email], " +
+                               "[Payer].[FirstName], " +
+                               "[Payer].[LastName], " +
+                               "[Payer].[Name] " +
+                               "FROM [payments].[Payers] AS [Payer] " +
+                               "WHERE [Payer].[Id] = @PayerId";
```

Comment:
The same as above - you need only the  email.

 Maybe create some static class PayerEmailProvider with static method which gets email based on id and use it in that 2 places (DRY)?

Reply:
Yup, good idea! Where such class should be placed?

Reply:
Maybe `Application/Payers` ?

Reply:
@kgrzybek please look at updated code :)